### DISCUSSION TITLE
Error running nextjs

### DISCUSSION THREAD

**Discussion by juanppdev** [vercel/next.js#47165] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
npm run dev                           
> dev                                                                                                                          > next dev

/data/data/com.termux/files/home/Fomundi/Cursos/nextjs/nextFolder/node_modules/next/dist/cli/next-dev.js:362
            showAll: args["--show-all"] ?? false,
                                         ^

SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.dev (/data/data/com.termux/files/home/Fomundi/Cursos/nextjs/nextFolder/node_modules/next/dist/lib/commands.js:10:30)
    at Object.<anonymous> (/data/data/com.termux/files/home/Fomundi/Cursos/nextjs/nextFolder/node_modules/next/dist/bin/next:141:28)
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

npm install next and npm run dev

### To Reproduce

I show everything in the code that I put

### Describe the Bug

I show everything in the code that I put

### Expected Behavior

I show everything in the code that I put

**Comment by JesseKoldewijn** [comment]
Could you maybe give us some more details? Like a link to a repo where you're encountering this issue, project environment etc.

**Comment by juanppdev** [comment]
For now I'm running it locally, and I don't know how I could give you more details

**Comment by JesseKoldewijn** [comment]
Maybe push it into a github repo or something like that? Otherwise replicating this will be kinda difficult.

**Comment by JesseKoldewijn** [comment]
You can get your environment details with the following command:
```sh
next info
```

This is also what is asked for in the issue template under "Provide environment information"

**Comment by valandex4** [comment, marked as answer]
What is your `node.js` version? 
You can find this using `node -v`, or, alternatively, `node --version`.

This `??` syntax is not supported in some older Node.js versions.

**Reply by balazsorban44** [reply]
Make sure you are on at least Node.js 14.

Ref:
https://node.green/#ES2020-features--nullish-coalescing-operator-----
https://nextjs.org/docs#system-requirements

**Comment by JoelBonet** [comment]
Probably @valandex4 is right (even though it's supported since Node 14 if I'm not wrong), if you want to do an extra test on that:

Could you please try to rewrite the statement to `showAll: args["--show-all"] || false` and tell if the issue persists?

**Sidenote**: the nullish coalescing operator(`??`) will fallback to the right statement when the first one is `null` or `undefined`, whereas the logical OR operator (`||`) will do so whenever the fist statement is `null`, `NaN`, `0`, empty string (`""` or `''` or ``) or, `undefined`, hence they are not always interchangeable. On the other hand, for the sake of unlocking this issue it may come in handy.

**Comment by iamdevlinph** [comment]
I opened a pull request https://github.com/vercel/next.js/pull/47250 to avoid this in the future
